### ISSUE TITLE
Kernel upload fail

### ISSUE THREAD

**Issue by ghost** [TeamShinkansen/Hakchi2-CE#162]
My Classic seems to be "stuck" between being able to play it and being able to flash it. When I turn it on normally, it's perfectly fine and playable, and I have retroarch installed, so it alternates fine between the original and RA menus, but when I try to flash the kernel, the flashing seems to "hang" and it stays stuck on the boot screen. Any attempt to flash, repair, uninstall, dump or browse all seem to hang, but the console is otherwise perfectly playable in its current state
![image](https://user-images.githubusercontent.com/43544902/48376983-25e24680-e692-11e8-90b1-e9751e14bda8.png)

**Comment by ghost** [comment]
I actually just solved my issue. It turns out that while I was running
RetroArch conversion as a main firmware, I wasn't shutting down
the SNES Classic properly. It's really easy. Just "Quit Retroarch" and go
back to the basic home menu for the system, then turn the system off
normally, then try to run Hakchi again through the PC program. As long as
you quit out of Retroarch properly, it should run and play every time.
Personally, I felt I didn't need RA as a base firmware and removed the
"Conversion" mod but still kept the "Neo" version so that it runs whenever
you play a game instead of swallowing up your main firmware. It's actually
deceptively easy once I gave it some thought. That's why the system
wouldn't turn off correctly when I hit the power switch while running RA
Conversion.

**Comment by boxed** [comment]
I have the same symptom butdont have retroarch installed so there must be something with your case that gives a hint about the wider problem. Maybe you should reopen this ticket.

**Comment by DanTheMan827** [comment]
So do you still have a problem or not?

**Comment by ghost** [comment]
Not anymore. Someone else talked me into keeping the topic open

**Comment by boxed** [comment]
Sorry, that was me. I forgot to update my comment.

I have the exact same error message, but I don't have retroarch. I installed hakchi when SNES classic was just released here in Sweden and now when I try to add a new game I get the message that I must upgrade the kernel but I get the error shown in this screenshot when I try. None of the options to repair, install, etc work.

**Comment by Walter-Maus** [comment]
I've the same problem as @boxed. I have bought the the SNES mini, installed the hakchi 3.5.2, installed the USB driver correctly and selected the "Kernel --> Install". I only get the same message as in the screenshot above.

**Comment by DanTheMan827** [comment]
This is almost always a firewall blocking hakchi2 CE or a VPN

Try allowing an exception for it in the firewall settings or just turn it off altogether as a test

**Comment by ghost** [comment]
Not with me, and I have the mother of all firewalls in effect. I've never
had to go online with Hakchi CE other than to update it

**Comment by littlebalup** [comment]
I have the same issue with the v3.5.2 on a stock console

edit: I switched from an USB3.0 port to an USB2.0 on my computer and it worked

**Comment by ARealityDivided** [comment]
**NES Classic/CLV S NESA USZ Model CLV-001
Hakchi-CE-3.5.2 Installed + Drivers on Windows 10.
Firewall: Off
Antivirus: Off
VPN: Off
USB3/USB2 Connections recognized. (No Difference)
Hakchi elevated to Administrator on execute. (No Difference)
Hakchi set to windows 8 compatibility. (No Difference)
Controller Attached and HDMI to monitor for output. (Stock boots fine)
Hakchi-CE-3.5.1 attempted for good measure. (No Difference)
Tried zadig. (No Difference)**

**Problem:
Hakchi Open>Install/Repair>Connect Device>Hold Reset>Press Power>Detected>Let Reset Go.**
Installer makes it to running a boota 474XXXXX command just after 22% and system forces into a reboot.
Progress bar states 28%, immediately system amber light is lit, unit is ejected from the computer and wait time runs out for Hakchi saying its taking longer than normal to boot.
Visually the system reboots during this time, however hangs at the Knock, Knock Hakchi boot screen and never boots nor reconnects itself back to the computer via usb. During this point it is also unresponsive to toggling the power button off and only way to shut down is to pull its usb connection where its drawing power. Machine will boot stock fine and be detected by windows. Any thoughts on what the problem might be?

**Solution: Click Start>Settings>Update&Security>Recovery>Advanced Startup+Reboot Now>TroubleShoot>Advanced Options>Startup Settings>F7 Option to Disable Driver Signature Enforcement** on reboot the attempt to install was successful.

**Comment by littlebalup** [comment]
> 
> 
> I have the same issue with the v3.5.2 on a stock console
> 
> edit: I switched from an USB3.0 port to an USB2.0 on my computer and it worked

more details:

When I plug it to my USB 3 port, Win7 detect it as "classic" and it doesn't work.
![image](https://user-images.githubusercontent.com/6732619/49252711-0e8eb180-f425-11e8-998a-8cda295c9ace.png)

When I plug it to USB 2 port, Win7 detect it as "Remote NDIS based Internet Sharing Device" and it works.
![capture2](https://user-images.githubusercontent.com/6732619/49252616-c66f8f00-f424-11e8-8470-84d499f079a2.PNG)

I can provide USB tree view report if needed. Hope it helps.


Edit : updating manually the driver when connected to USB 3 works:
Right click to the "classic" device in the device manager -> update driver software -> browse my computer... -> let me pick from list ... -> network adapters -> Microsoft Corporation -> Remote NDIS based Internet Sharing Device

Edit 2 : I finally ran the windows "troubleshoot" procedure to "configure a device" just in case : https://support.microsoft.com/en-us/help/2970908/how-to-use-microsoft-easy-fix-solutions
It found that I disabled driver update from windows update, so re-enabled it then seems it installed a Samsung driver (see next picture). Now when I plug the mini to my USB3, the "Remote NDIS based Internet Sharing Device" is auto-detected and all seems to works as it should.
![image](https://user-images.githubusercontent.com/6732619/49259042-08560080-f438-11e8-9299-740911628764.png)
